src/MiniCover/Instrumentation/CustomAssemblyResolver.cs

```diff
@@ -25,22 +46,37 @@ AssemblyDefinition GetAssembly(string file, ReaderParameters parameters)
 
         protected override AssemblyDefinition SearchDirectory(AssemblyNameReference name, IEnumerable<string> directories, ReaderParameters parameters)
         {
-            foreach (var directory in directories)
+            if (_dependencyContext != null)
             {
-                var files = FileUtils.GetFiles(null, null, $"**/{name.Name}.dll", directory);
-                foreach (var file in files)
+                var dependency = _dependencyContext.CompileLibraries.FirstOrDefault(c =>
                 {
-                    try
-                    {
-                        return GetAssembly(file, parameters);
-                    }
-                    catch (BadImageFormatException)
+                    return c.Name == name.Name;
+                });
+
+                if (dependency != null)
+                {
+                    foreach (var depAssembly in dependency.Assemblies)
                     {
-                        continue;
+                        foreach (var p in directories)
+                        {
+                            var file = Path.Combine(new[] { p, dependency.Path, depAssembly }.Where(x => x != null).ToArray());
```

Comment:
This is where the assembly version is validated.
`p` is one of the search directories. Eventually it will be `~/nuget/packages`.
`dependency.Path` is something like `somePackage/1.2.1`
`depAssembly` should be something like `lib/netstandard2.0/SomePackage.dll`

When I combined all that info I got:
`~/.nuget/packages/SomePackage/1.2.1/lib/netstandard2.0/SomePackage.dll`.

But to be honest I don't know if it works well, I need you to try it, or maybe share a sample project with me that has the error, so I can also test it.

Reply:
This is kind of a reverse engineering + guess on how .NET Core uses *deps.json* and *runtime.config.json*.